arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token);
+          token.arabicFontLang = 'ar';
+          token.data[0].data[0] = mapped;
+        }
+
+        return token;
+      },
+      _getArgumentMML: function (name) {
+        //  returns an argument that is a single MathML element
+        //  (in an mrow if necessary)
+        //
+        //  This functions has been copied here from extensions/TeX/HTML.js, to avoid
+        //  adding a dependency.
+        //
+        //  TODO: Consider importing (as a dependency) this from HTML.js instead!
+        var arg = this.ParseArg(name);
+        if (arg.inferred && arg.data.length === 1) {
+          arg = arg.data[0];
+        } else {
+          delete arg.inferred;
+        }
+
+        return arg;
+      },
+      Push: function () {
+        var beforeLang = this.stack.env.lang;
+        var retVal = texParsePush.apply(this, arguments);
+
+        // Huge giant hack to propagate `lang` from Arrays (and other environments?)
+        // to their children fractions and others.
+        if (beforeLang) {
+          if (!this.stack.env.lang) {
+            this.stack.env.lang = beforeLang;
+          }
+        }
+
+        return retVal;
+      },
+      mmlToken: function (token) {
+        // TODO: Check for possible incompatibility with boldsymbol extension
+        var parsedToken = texParseMMLToken.call(this, token);
+
+        if ('ar' === this.stack.env.lang) {
+          this.markArabicToken(parsedToken);
+        }
+
+        return parsedToken;
+      },
+      markArabicToken: function (token) {
+        if ('mn' === token.type) {
+          return this.arabicNumber(token);
+        } else if ('mi' === token.type) {
+          return this.arabicIdentifier(token);
+        } else if ('mo' === token.type) {
+          return this.arabicOperator(token);
+        }
+
+        return token;
+      },
+      HandleArabic: function (name) {
+        if (MathJax.Hub.config.Arabic.isArabicPage) {
+          this.MarkAsArabic(name);
+        }
+      },
+      TranslateTeX: function (name) {
+        var english = this.GetArgument(name);
+        var arabicText = this.GetArgument(name);
+        var helper = Arabic.TeX(english, arabicText);
+        return helper.call(this, name);
+      },
+      TranslateText: function (name) {
+        var english = this.GetArgument(name);
+        var arabicText = this.GetArgument(name);
+        var helper = Arabic.Text(english, arabicText);
+        return helper.call(this, name);
+      },
+      TranslateSymbols: function (name) {
+        var english = this.GetArgument(name);
+        var arabicText = this.GetArgument(name);
+        var helper = Arabic.Symbols(english, arabicText);
+        return helper.call(this, name);
+      },
+      MarkAsArabic: function (name) {
+        this.stack.env.lang = 'ar';
+        var arg = this._getArgumentMML(name);
+        this.Push(this.flipHorizontal(arg));
+      },
```

Comment:
This function leaves `this.stack.env.lang` set to `'ar'`, so this would mean that everything after the use of `\ar{}` or `\alwyasar{}` will always be in arabic.  I.e., `\alwaysar{x=1} x=1` end up producing

![arabic-cibara](https://cloud.githubusercontent.com/assets/432782/15217820/41653878-182b-11e6-950f-0fce83574e5a.png)

for me.

The solution is to save the old setting of `lang` and put it back after the typesetting is done.

```
MarkAsArabic: function (name) {
  var lang = this.stack.env.lang;
  this.stack.env.lang = 'ar';
  var arg = this._getArgumentMML(name);
  this.stack.env.lang = lang;
  this.Push(this.flipHorizontal(arg));
},